### PR TITLE
autosave funcs, modules and clusters

### PR DESCRIPTION
**Pull request by jlewitt1**

### CODE REVIEW

**Comment by jlewitt1** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1188?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1188** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1188?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1187** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1187?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jlewitt1 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: runhouse/resources/functions/function_factory.py [hunk 1]
```diff
@@ -121,4 +122,7 @@ def function(
     if load_secrets and not dryrun:
         new_function.send_secrets()
 
+    if rns_client.save_to_den and rns_client.token:
```
Comment (dongreenberg) [inline runhouse/resources/functions/function_factory.py:125]: Should we centralize this in an rns_client.autosave_resources() method?

File: .github/workflows/setup_rh_config/action.yaml [hunk 2]
```diff
@@ -27,3 +27,4 @@ runs:
         echo "token: ${{ inputs.token }}" >> ~/.rh/config.yaml
         echo "username: ${{ inputs.username }}" >> ~/.rh/config.yaml
         echo "api_server_url: ${{ inputs.api_server_url }}" >> ~/.rh/config.yaml
+        echo "autosave: false" >> ~/.rh/config.yaml
```
Comment (mkandler) [inline .github/workflows/setup_rh_config/action.yaml:30]: Just a question: could it make sense to have autosave possibly be an array of types to autosave? e.g. `['cluster', 'module']` or is that too much control? Thinking of cases where some people might want to autosave functions as well, or maybe autosave everything (envs, ...)?

Comment (jlewitt1) [reply .github/workflows/setup_rh_config/action.yaml:30]: for envs i think less relevant, bc those get saved automatically as subresources of functions or modules, maybe clusters could be treated differently, but i think that's adding more complexity than we need for now unless we get feedback about having more fine grained control over the resource types

Comment (rohinb2) [reply .github/workflows/setup_rh_config/action.yaml:30]: yeah i agree that we can jsut leave it as a generic param

**Review by rohinb2 — APPROVED** [review]

**Pull request merged**